File: 242-valid-anagram/242-valid-anagram.cpp
#include <unordered_map>
#include <string>

class Solution {
 public:
  std::unordered_map<char, int> getCharCounts(const std::string& s) {
    std::unordered_map<char, int> char_counts;
    for (const char& c : s) {
      char_counts[c]++;
    }
    return char_counts;
  }

  bool isAnagram(std::string s, std::string t) {
    if (s.length() != t.length()) {
      return false;
    }
    return getCharCounts(s) == getCharCounts(t);
  }
};

File: 242-valid-anagram/242-valid-anagram.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public Map<Character, Integer> getCharCounts(String s) {
        Map<Character, Integer> charCounts = new HashMap<>();
        for (Character ch : s.toCharArray()) {
            charCounts.put(ch, charCounts.getOrDefault(ch, 0) + 1);
        }
        return charCounts;
    }
    
    public boolean isAnagram(String s, String t) {
        return getCharCounts(s).equals(getCharCounts(t));
    }
}